### PR TITLE
feat: make urlSource compatible with new ipfs.add

### PR DESCRIPTION
**Pull request by Gozala**
fixes: ipfs/js-ipfs#3195

### CODE REVIEW

**Comment by Gozala** [comment]
@achingbrain I'm unable to push a branch to this repo, so I had to use my own fork for this pull request.

File: src/files/url-source.js [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
-    content: response.iterator()
+    content: readURLContent(url, options)
   }
 }
+
+const readURLContent = async function * (url, options) {
+  const http = new Http()
+  const response = await http.get(url, options)
+  yield * response.iterator()
+}
```
Comment (hugomrdias) [inline src/files/url-source.js:16]: i dont understand this change, why the wrapper ?

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: src/files/url-source.js (hunk 1 above)
Comment (Gozala) [reply src/files/url-source.js:16]: @hugomrdias so previously we had `ipfs.add` that could imported bunch of files or import a single one. Which later we have split into `ipfs.add` that imports a single file and takes single file like thing, and `ipfs.addAll` which imports multilpe files.

This change just makes return of `urlSource` compatible with `ipfs.add` as opposed to `ipfs.addAll`.

P.S.: `ipfs.add` implementation (but not the typings) will still accept many files, but plan is to make that not a case in the future.

Comment (Gozala) [reply src/files/url-source.js:16]: To answer your specific question, this does not add a wrapper but rather moves code doing the content fetching into `content` field which is expected to be `AsyncIterable<Uint8Array>`. effectively changing return type of this `urlSource` from `AsyncIterable<{ path: string, content: AsyncIterable<Uint8Array> }>` to `{ path: string, content: AsyncIterable<Uint8Array> }` removing one layer of asynchronicity and making return be a single file as opposed to collection of files that always happen to contain one.

**Review by achingbrain — APPROVED** [review]

**Pull request merged**